File: katt/magic/attachments/python2/grader.py
import grader
import magic
import sys

_tricks = []
def _grader_init():
    N, K = map(int, sys.stdin.readline().split())
    L = list(map(int, sys.stdin.readline().split()))
    R = list(map(int, sys.stdin.readline().split()))
    res = magic.magic_score(N, K, L, R)
    if len(_tricks) != N:
        print("error: tried to perform {} tricks, should perform {}".format(len(_tricks)), N)
    else:
        print(res)
        print(" ".join(map(str, _tricks)))

def trick(ans):
    _tricks.append(ans)

if __name__ != "__main__":
    grader._grader_init()

File: katt/Contest1/magic/attachments/cpp/grader.cpp
#include "magic.h"
#include <cstdio>
#include <cassert>
#include <tuple>
#include <vector>

using namespace std;

static std::vector<int> output;
void trick(int rabbits) {
	output.push_back(rabbits);
}

int main() {
	int N, K;
	ignore = scanf("%d%d", &N, &K);
	std::vector<int> L(N), R(N);
	for (int i = 0; i < N; i++)
		ignore = scanf("%d", &L[i]);
	for (int i = 0; i < N; i++)
		ignore = scanf("%d", &R[i]);
	output.reserve(N);
	int sol = magic_score(N, K, L.data(), R.data());
	if ((int)output.size() != N) {
		printf("error: tried to perform %d tricks, should perform %d", (int)output.size(), N);
	} else {
		printf("%d\n", sol);
		for (int i = 0; i < N; i++) {
			if (i != 0) putchar(' ');
			printf("%d", output[i]);
		}
		puts("");
	}
}
